packages/model/addon/-private/model.js

```diff
@@ -29,38 +26,15 @@ import {
   InternalModel,
   PromiseObject,
   recordDataFor,
-  recordIdentifierFor,
 } from '@ember-data/store/-private';
 
 import Errors from './errors';
+import RecordState, { peekTag, tagged } from './record-state';
 import { relationshipFromMeta } from './system/relationships/relationship-meta';
 
-const RecordMeta = new WeakMap();
-function getRecordMeta(record) {
-  let meta = RecordMeta.get(record);
-  if (meta === undefined) {
-    meta = Object.create(null);
-    RecordMeta.set(record, meta);
-    if (DEBUG) {
-      record.____recordMetaCache = meta;
-    }
-  }
-  return meta;
-}
-
-function getWithDefault(meta, prop, value) {
-  let v = meta[prop];
-  if (v === undefined) {
-    return value;
-  }
-  return v;
-}
-
 const { changeProperties } = Ember;
 
-function isInvalidError(error) {
-  return error && error.isAdapterError === true && error.code === 'InvalidError';
-}
+const flagToggledDecorator = CUSTOM_MODEL_CLASS ? dependentKeyCompat : tagged;
```

Comment:
Can you please add a comment describing the goal/purpose of this.

Reply:
sure! the TL;DR is that in the non-CUSTOM_MODEL_CLASS world things decorated with `flagToggledDecorator` have logic that requires a cache and caching key (which tagged provides), but in the CUSTOM_MODEL_CLASS branch it only requires `dependentKeyCompat` which ensures computeds/observers on the field will fire when anything it accesses that is tracked changes. I'll add this to the PR.